apps/alert_processor/lib/model/metadata.ex

```diff
@@ -0,0 +1,46 @@
+defmodule AlertProcessor.Model.Metadata do
+  @moduledoc """
+  Generic schema that allows using the database as a key-value store.
+
+  Note: Does not provide conveniences for handling concurrent access, as the only use of this
+  schema currently is within a database-level exclusive lock (see `AlertProcessor.Lock`).
+  """
+
+  @type key :: atom
+  @type value :: map
+
+  use Ecto.Schema
+  import Ecto.Query
+  alias AlertProcessor.Repo
+
+  @primary_key {:id, AlertProcessor.AtomType, autogenerate: false}
+
+  schema "metadata" do
+    field(:data, :map, default: %{}, null: false)
+    timestamps()
+  end
+
+  @spec get(key) :: value
+  def get(id) when is_atom(id) do
```

Comment:
It seems like these 3 functions could use some quick tests?

Reply:
Yes, they could! Forgot all about it. 🙃